e2e/stack_monitoring_test.go

```diff
@@ -0,0 +1,614 @@
+		}
+
+		err = serviceManager.RunCommand("stack-monitoring", composes, []string{"logs", "metricbeat"}, env)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"error":             err,
+				"metricbeatVersion": stackVersion,
+				"product":           sm.Product,
+				"serviceVersion":    stackVersion,
+			}).Error("Could not retrieve Metricbeat logs")
+
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (sm *StackMonitoringTestSuite) runProduct(product string, collectionMethod string) error {
+	env := map[string]string{
+		"logLevel":       log.GetLevel().String(),
+		product + "Tag":  stackVersion, // all products follow stack version
+		product + "Port": sm.Port,      // we could run the service in another port
+		"stackVersion":   stackVersion,
+	}
+	env = config.PutServiceEnvironment(env, product, stackVersion)
+
+	for k, v := range sm.Env {
+		env[k] = v
+	}
+
+	log.Debugf("Installing %s", sm.Product)
+	srvManager := services.NewServiceManager()
+
+	err := srvManager.AddServicesToCompose("stack-monitoring", []string{product}, env)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"port":    sm.Port,
+			"service": product,
+			"version": stackVersion,
+		}).Error("Could not run the service.")
+		return err
+	}
+
+	if collectionMethod == "metricbeat" {
+		err = sm.runMetricbeat()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// @product the product to be installed. Valid values: elasticsearch, kibana, beats, logstash
+// @collectionMethod the collection method to be used. Valid values: legacy, metricbeat
+func (sm *StackMonitoringTestSuite) sendsMetricsToElasticsearch(
+	product string, collectionMethod string) error {
+
+	// validate product
+	err := sm.checkProduct(product, collectionMethod)
+	if err != nil {
+		return err
+	}
+
+	err = sm.runProduct(product, collectionMethod)
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Running %[1]s for X seconds (default: 30) to collect monitoring data internally and index it into the Monitoring index for %[1]s", product)
+	sleep("30")
+
+	composes := []string{
+		sm.Product, // product service
+	}
+
+	if collectionMethod == "metricbeat" {
+		composes = append(composes, "metricbeat")
+	}
+
+	log.Debugf("Stopping %s", product)
+	srvManager := services.NewServiceManager()
+
+	err = srvManager.RemoveServicesFromCompose("stack-monitoring", composes, sm.Env)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error":   err,
+			"service": product,
+			"version": stackVersion,
+		}).Error("Could not stop the service.")
+		return err
+	}
+
+	log.Debugf("Downloading sample documents from %s's monitoring index to a test directory", product)
+	hits, err := sm.getCollectionMethodHits()
+	if err != nil {
+		return err
+	}
+
+	sm.collectionHits[collectionMethod] = hits
+	log.Debugf("Hits: %v", hits)
+
+	log.Debugf("Deleting monitoring index %s", sm.IndexName)
+	fn := func(ctx context.Context) {
+		err := deleteIndex(ctx, sm.IndexName)
+		if err != nil {
```

Comment:
Nit: since `deleteIndex` just returns an error, you could collapse these two statements into:

```
if deleteIndex(...) != nil {
   ...
}
```

or, alternatively if you prefer having the `err` around:

```
if err := deleteIndex(...); err != nil {
```